File: android/src/main/java/com/rozetkapaysdk/converters/ConfigConverters.kt
package com.rozetkapaysdk.converters

import com.rozetkapay.sdk.init.RozetkaPaySdkMode

fun String.toRozetkaPaySdkMode(): RozetkaPaySdkMode {
  return when (this.lowercase().trim()) {
    "production" -> RozetkaPaySdkMode.Production
    "development" -> RozetkaPaySdkMode.Development
    else -> throw IllegalArgumentException("Unknown RozetkaPaySdk mode: $this")
  }
}

File: ios/ConfigConverters.swift
import Foundation
import RozetkaPaySDK

extension String {
    func toRozetkaPaySdkMode() throws -> RozetkaPaySdkMode {
        switch self.lowercased().trimmingCharacters(in: .whitespacesAndNewlines) {
        case "production":
            return .production
        case "development":
            return .development
        default:
            throw NSError(domain: "RozetkaPaySdk", code: -1, userInfo: [NSLocalizedDescriptionKey: "Unknown RozetkaPaySdk mode: \(self)"])
        }
    }
}

extension Optional where Wrapped == String {
    func toFieldRequirement(defaultValue: FieldRequirement = .none) -> FieldRequirement {
        switch self?.lowercased().trimmingCharacters(in: .whitespacesAndNewlines) {
        case "required":
            return .required
        case "optional":
            return .optional
        case "none":
            return .none
        default:
            return defaultValue
        }
    }
}
